lib/SyTest/Homeserver/Synapse.pm

```diff
@@ -202,11 +201,13 @@ sub start
         "listeners" => $listeners,
 
         "bcrypt_rounds" => 0,
-        "start_pushers" => (not $self->{pusher}),
 
-        "notify_appservices" => (not $self->{appservice}),
-
-        "send_federation" => (not $self->{federation_sender}),
+        # If we're using dendron-style split workers, we need to disable these
+        # things in the main process
+        "start_pushers"      => ( not $self->{dendron} ),
+        "notify_appservices" => ( not $self->{dendron} ),
+        "send_federation"    => ( not $self->{dendron} ),
+        "notify_appservices" => ( not $self->{dendron} ),
```

Comment:
You seem to have `"notify_appservices" => ( not $self->{dendron} ),` in there twice.